mlx/io/load.h

```diff
@@ -4,6 +4,7 @@
 
 #include <memory>
 #include <sstream>
+#include <variant>
```

Comment:
Not critical (maybe overkill).. but I think it would make sense to reorganize:
- move the json declarations into mlx/io/json.h (and include this for json only)
- move the json definitions into mlx/io/json.cpp

Reply:
Sounds great!